components/nodemanager-service/pgdb/filtering.go

```diff
@@ -38,31 +38,31 @@ func mergeFilters(mergeableFilters []*common.Filter) ([]common.Filter, error) {
 	return filters, nil
 }
 
-func handleTagFilters(tagFilters []common.Filter) (string, error) {
-	var tagConditions []string
+func handleTagFilters(tagFilters []common.Filter) ([]string, error) {
+	tagConditions := make([]string, 0)
```

Comment:
Minor, we could pre-allocate this array here:

```suggestion
	tagConditions := make([]string, 0, len(tagFilters))
```

Reply:
Thank you Steven, I pushed a commit with your suggestions.